### DISCUSSION TITLE
ConnectionRefusedError: [Errno 111] Connection refused - Docker

### DISCUSSION THREAD

**Discussion by paulgibbins** [blakeblackshear/frigate#398] [General]
Hi - Hope someone may be able to advise please. I am trying to run Frigate to monitor objects on my Reolink cameras but can't get the container to load. After running 'docker-compose up -d' I see the container restart after a few seconds. When I check the Docker logs I see:
Traceback (most recent call last):
  File "detect_objects.py", line 441, in <module>
    main()
  File "detect_objects.py", line 152, in main
    client.connect(MQTT_HOST, MQTT_PORT, 60)
  File "/usr/local/lib/python3.8/dist-packages/paho/mqtt/client.py", line 941, in connect
    return self.reconnect()
  File "/usr/local/lib/python3.8/dist-packages/paho/mqtt/client.py", line 1075, in reconnect
    sock = self._create_socket_connection()
  File "/usr/local/lib/python3.8/dist-packages/paho/mqtt/client.py", line 3546, in _create_socket_connection
    return socket.create_connection(addr, source_address=source, timeout=self._keepalive)
  File "/usr/lib/python3.8/socket.py", line 808, in create_connection
    raise err
  File "/usr/lib/python3.8/socket.py", line 796, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

My docker-compose.yml looks like this:

frigate:
  container_name: frigate
  restart: unless-stopped
  privileged: true
  shm_size: '1g'
  image: blakeblackshear/frigate:stable-amd64
  volumes:
    - /etc/localtime:/etc/localtime:ro
    - /volume1/surveillance:/config
    - /volume1/surveillance:/clips
  ports:
    - "5555:5555"

My config.yml looks like this:
web_port: 5555

mqtt:
  host: 192.168.86.61
  topic_prefix: tensorflow
  client_id: tensorflow
  user: mqtt_tensorflow
  password: ***********

save_clips:
  max_seconds: 300

cameras:
  Patio:
    ffmpeg:
      input: rtsp://admin:*************@192.168.86.39:554/h264Preview_01_main
    take_frame: 2
    best_image_timeout: 60

    objects:
      track:
        - person
        - car

    mqtt:
      snapshot_height: 480
      crop_to_region: True

    save_clips:
      enabled: True
      pre_capture: 30

    snapshots:
      show_timestamp: False
      draw_zones: True

I am able to connect to the camera via VLC using rtsp://admin:*************@192.168.86.39:554/h264Preview_01_main
The config I am using is from someone I know who uses it to connect to their same Reolink cameras, obviously I just changed the details specific to my set up.

Appreciate any guidance you may be able to give me

**Comment by hawkeye217** [comment]
Looks like you need to double check your MQTT server details. The error is related to that.
